### ISSUE TITLE
Dollar sign ($) in .env value converted to 'undefined' in $config

### ISSUE THREAD

**Issue by gbruins-notours** [nuxt/nuxt.js#8023]
### Versions

- nuxt: v2.14.4
- node: v12.14.1

### Reproduction

<details open>
<summary>Additional Details</summary>
<br>

</details>

### Steps to reproduce
- In a .env file, add an entry that contains a $ in the value:
```
MY_PASSWORD=bug$
FOO=$bug
```

- At the top of your nuxt.config.js file, import dotenv:  
```
require('dotenv').config();
```

- In nuxt.config.js, set a privateRuntimeConfig attribute, using the .env value:
```
privateRuntimeConfig: {
  myPassword: process.env.MY_PASSWORD,
  foo: process.env.FOO
}
```

### What is Expected?
context.$config.myPassword should equal 'bug$'
context.$config.foo should equal '$bug'

### What is actually happening?
context.$config.myPassword equals 'bugundefined'
context.$config.foo equals undefined    (NOTE: not a string! This is the javascript value undefined)

This means that any value in .env that contains a '$' character wont work in $config, which seems highly likely given that you're probably keeping passwords, api keys, etc in .env

**Comment by gbruins-notours** [comment]
Perhaps related to https://github.com/nuxt/nuxt.js/issues/7990

**Comment by gbruins-notours** [comment]
Workaround: 
Escaping your values in .env with a double backslash will produce the expected result in $config
```
MY_PASSWORD=bug\\$
```

**Comment by pi0** [comment]
Hi @gbruins-notours It should be fixed by #8013 (you can try with using `nuxt-edge` package). But escaping $ is best workaround.
